h/api/logic.py

```diff
@@ -51,20 +48,12 @@ def _anonymize_deletes(annotation):
         annotation['permissions'][action] = filtered
 
 
-def update_annotation(annotation, fields, has_admin_permission):
+def update_annotation(annotation, fields):
+    """Update the given annotation with the given new fields."""
     # Some fields are not to be set by the user, ignore them
     for field in PROTECTED_FIELDS:
         fields.pop(field, None)
 
-    # If the user is changing access permissions, check if it's allowed.
-    changing_permissions = (
-        'permissions' in fields and
-        fields['permissions'] != annotation.get('permissions', {})
-    )
-    if changing_permissions and not has_admin_permission:
-        raise RuntimeError("Not authorized to change annotation permissions.",
-                           401)  # Unauthorized
```

Comment:
Nice that `validate_permissions()` makes this check, which didn't really fit into the Pyramid auth system, no longer needed.